### ISSUE TITLE
No warning or error if an invalid target attribute is set to None

### ISSUE THREAD

**Issue by jmillikin-stripe** [bazelbuild/bazel#11000]
### Description of the problem / feature request:

Given the following BUILD file:
```python
cc_binary(
    name = "hello",            
    srcs = ["hello.c"],        
    no_such_attribute = None,
)
```

I would expect some sort of error to be printed. There _is_ an error if the invalid attribute is set to some non-`None` value (`0`, `False`, whatever) but `None` is just silently ignored.

A warning would also be acceptable (backwards compatibility and all), maybe with a flag to make it a hard error (defaulting to `true` in the next breaking release).

This behavior happens for both built-in and Starlark-defined rules.

### What operating system are you running Bazel on?

macOS

### What's the output of `bazel info release`?

release 2.2.0

**Comment by alandonovan** [comment]
Dup of Google-internal issue: b/122552411.

**Comment by jin** [comment]
Is this still a P1 bug?

**Comment by brandjon** [comment]
This is certainly surprising, and should be an error for non-existent attributes.

Should `None` be a valid value for attributes that do exist though? For instance, I could imagine it having the same effect as omitting an optional attribute of Label type. Indeed, this might be necessary in order to support feeding the output of `native.existing_rule` back into a target definition. It'd presumably still be an error for mandatory attributes, and perhaps list-typed attributes that can use `[]` as the zero value.

**Comment by comius** [comment]
This was fixed.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 6.4.0 RC1](https://releases.bazel.build/6.4.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
